src/System.IO.StreamSourceGeneration/src/StreamSourceGen.cs

```diff
@@ -0,0 +1,177 @@
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace System.IO.StreamSourceGeneration;
+
+[Generator]
+public partial class StreamSourceGen : IIncrementalGenerator
+{
+    internal const string StreamBoilerplateAttributeFullName = "System.IO.StreamSourceGeneration.GenerateStreamBoilerplateAttribute";
+    internal const string StreamFullName = "System.IO.Stream";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(ctx =>
+        {
+            //Diagnostics.Debugger.Launch();
+            ctx.AddSource("TaskToApm.g.cs", System.IO.StreamSourceGeneration.Properties.Resources.TaskToApm);
+        });
+
+        IncrementalValuesProvider<ClassDeclarationSyntax> classDeclarations = context.SyntaxProvider
+            .ForAttributeWithMetadataName(
+                StreamBoilerplateAttributeFullName,
+                predicate: (node, _) => node is ClassDeclarationSyntax c && c.Modifiers.Any(SyntaxKind.PartialKeyword),
+                transform: (context, _) => (ClassDeclarationSyntax)context.TargetNode);
+
+        IncrementalValueProvider<(Compilation, ImmutableArray<ClassDeclarationSyntax>)> compilationAndClasses =
+            context.CompilationProvider.Combine(classDeclarations.Collect());
+
+        context.RegisterSourceOutput(compilationAndClasses, (spc, source) => Execute(source.Item1, source.Item2, spc));
+    }
+
+    private static void Execute(Compilation compilation, ImmutableArray<ClassDeclarationSyntax> classes, SourceProductionContext context)
+    {
+        if (classes.IsDefaultOrEmpty)
+        {
+            return;
+        }
+
+#if LAUNCH_DEBUGGER
+        if (!Diagnostics.Debugger.IsAttached)
+        {
+            Diagnostics.Debugger.Launch();
+        }
+#endif
+
+        List<StreamTypeInfo>? classesWithGenerationOptions = Helpers.GetClassesWithGenerationOptions(compilation, classes, context.CancellationToken);
+
+        if (classesWithGenerationOptions == null)
+        {
+            return;
+        }
+
+        StringBuilder sb = new();
+
+        foreach (StreamTypeInfo streamTypeInfo in classesWithGenerationOptions)
+        {
+            sb.Clear();
+            Emit(sb, streamTypeInfo);
+
+            INamedTypeSymbol typeSymbol = streamTypeInfo.TypeSymbol;
+            string hintName = $"{typeSymbol.ContainingNamespace}.{typeSymbol.Name}.Boilerplate.g.cs";
+            context.AddSource(hintName, sb.ToString());
+        }
+    }
+
+    private static void Emit(StringBuilder sb, StreamTypeInfo streamTypeInfo)
+    {
+        sb.Append($@"// <auto-generated/>
+
+#nullable enable
+
+{StreamBoilerplateConstants.UsingDirectives}
+
+namespace {streamTypeInfo.TypeSymbol.ContainingNamespace}
+{{");
+        sb.Append($@"
+    partial class {streamTypeInfo.TypeSymbol.Name}
+    {{");
+
+        HashSet<string> overriddenMembers = streamTypeInfo.OverriddenMembers;
```

Comment:
Since there aren't a lot of methods, and strings are expensive, I wonder if this wouldn't be better off as just having a struct for the overridden members, with it having a field for each one that we care about.

```C#
internal struct OverriddenMembers
{
    internal SomeWordHere ReadArray;
    internal SomeWordHere ReadSpan;
    ...
}
```

Or, a bit more loosey-goosey, an array indexed by named positions.

Reply:
I switched to a `HashSet<StreamMember>`, where StreamMember is an enum, let me know if that still concerns you.